### PR TITLE
Policies: Adds deprecation policy

### PR DESCRIPTION
**Pull request by bergquist**
Based on the internal document that have been circulated for a while. 

I'm not sure what the best place would be but I figure this belongs more under contributing than docs? But more than happy to change it.

### CODE REVIEW

File: contribute/deprecation-policy.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+| ------ | ---------- | ---------------------------------------------------------------- |
+| Large  | 1-2 years  | Classic alerting, scripted dashboards, AngularJS                 |
+| Medium | 6 months   | Supported Database for Grafana's backend                         |
+| Small  | 1-3 months | Refresh OAuth access_token automatically using the refresh_token |
+
+## Announced deprecations.
+
+| Name                  | Annoucement Date | Disabling date | Removal Date | Description                                                                                                               | Status  |
+| --------------------- | ---------------- | -------------- | ------------ | ------------------------------------------------------------------------------------------------------------------------- | ------- |
+| Support for Mysql 5.7 | 2023-05-15       | October 2023   |              | MySQL 5.7 is getting deprecated in October 2023 and Grafana's policy is to test against the officially supported version. | Planned |
```
Comment (bergquist) [inline contribute/deprecation-policy.md:32]: I'll write up an issue and link to it before merging.

File: contribute/deprecation-policy.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Deprecation policy
+
+Prior to Grafana 10, major releases were an opportunity to make breaking changes and remove code we didn’t want to maintain.
+
+Going forward that is something we will avoid in order to make it easier to upgrade Grafana. From now on we should avoid making unexpected breaking changes to the customer experience and our ability to operate our service.
+
+If we cannot remove the features without introducing a breaking chance we should follow an deprecation plan as listed below.
```
Comment (DanCech) [inline contribute/deprecation-policy.md:7]:
```suggestion
If we cannot remove a feature without introducing a breaking chance we should create a deprecation plan as listed below:
```

File: contribute/deprecation-policy.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Deprecation policy
+
+Prior to Grafana 10, major releases were an opportunity to make breaking changes and remove code we didn’t want to maintain.
+
+Going forward that is something we will avoid in order to make it easier to upgrade Grafana. From now on we should avoid making unexpected breaking changes to the customer experience and our ability to operate our service.
```
Comment (DanCech) [inline contribute/deprecation-policy.md:5]:
```suggestion
Going forward we will do our best to **not** make breaking changes, in order to make upgrading Grafana as easy and reliable as possible. From now on we will avoid making unexpected breaking changes to the customer experience and our ability to operate our service.
```

File: contribute/deprecation-policy.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+If we cannot remove the features without introducing a breaking chance we should follow an deprecation plan as listed below.
+
+- A deprecation plan should have the following steps.
+- Figure out usage stats for the feature.
```
Comment (DanCech) [inline contribute/deprecation-policy.md:10]:
```suggestion
- Determine usage levels of the feature.
```

**Review by DanCech — COMMENTED** [review]
I like where this is headed! Added a few suggestions

File: contribute/deprecation-policy.md (hunk 3 above)
Comment (bergquist) [reply contribute/deprecation-policy.md:5]: kind of passive voice 🤔 
yours is better but I might take another stab at this one

File: contribute/deprecation-policy.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# Deprecation policy
+
+Prior to Grafana 10, major releases were an opportunity to make breaking changes and remove code we didn’t want to maintain.
+
+Going forward we will do our best to **not** make breaking changes, in order to make upgrading Grafana as easy and reliable as possible. From now on we will avoid making unexpected breaking changes to the customer experience and our ability to operate our service.
+
+If we cannot remove a feature without introducing a breaking chance we should create a deprecation plan as listed below:
```
Comment (DanCech) [inline contribute/deprecation-policy.md:7]: Typo fix and stronger commitment
```suggestion
If we cannot remove a feature without introducing a breaking change we will create a deprecation plan as listed below:
```

File: contribute/deprecation-policy.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+- Migrate users if possible
+- Give users time to adjust to the deprecation.
+- Disable the feature by default.
+- Remove the feature from the code base.
+
+Depending on the size and importance of the feature this can be a design doc or an issue. We want this to be written communication for all parties so we know it's intentional and that did a reasonable attempt to avoid breaking changes unless needed. The size of the feature also means different notice times between Depreciation and disabling as well as disabling and removal. The actual duration will depend on releases of Grafana and the table below should be used as a guide.
+
+Grafana employees can find more details in our internal docs.
+
+## Grace period between announcement and disablement
```
Comment (DanCech) [inline contribute/deprecation-policy.md:22]: Not sure disablement is a word
```suggestion
## Grace period between announcement and disabling feature by default
```

**Review by DanCech — APPROVED** [review]
LGTM, if I were to add anything it would be guidance about the time between disable-by-default and full removal.

File: contribute/deprecation-policy.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
+# Deprecation policy
+
+Prior to Grafana 10, major releases were an opportunity to make breaking changes and remove code we didn’t want to maintain.
+
+Going forward we will do our best to **not** make breaking changes, in order to make upgrading Grafana as easy and reliable as possible. From now on we will avoid making unexpected breaking changes to the customer experience and our ability to operate our service.
```
Comment (torkelo) [inline contribute/deprecation-policy.md:5]: This has always from day 1 been a major goal and focus (to make upgrading as easy and as pain-free as possible). So not sure what is changing (other than adding a formal policy). This seems to conflate unexpected breaking changes and deprecating features. 

I would rephrase this a bit. 

```suggestion
Major releases are when we will be making breaking changes, like changing behavior or removing a feature. 

We always do our best **not** to introduce breaking changes in order to make upgrading Grafana as easy and reliable as possible. However, at times we have to introduce a breaking change by changing behavior or by removing a feature.  

But in order to remove a feature and introduce a breaking change we need to create a deprecation plan. 
```

File: contribute/deprecation-policy.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
+# Deprecation policy
+
+We do our best to limit breaking changes and the deprecation of features to major releases. We always do our best **not** to introduce breaking changes in order to make upgrading Grafana as easy and reliable as possible. However, at times we have to introduce a breaking change by changing behavior or by removing a feature.
+
+To minimize the negative effects of removing a feature we require a deprecation plan that includes:
+
+If we cannot remove a feature without introducing a breaking change we will create a deprecation plan as listed below:
```
Comment (torkelo) [inline contribute/deprecation-policy.md:7]: nitpick: These two sentences seem to say almost the thing and make it seem like a list will follow.

File: contribute/deprecation-policy.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+- A deprecation plan should have the following steps.
```
Comment (torkelo) [inline contribute/deprecation-policy.md:9]: This is the third "a deprecation plan that includes" in row :smile:   (see the two sentences above)

**Review by torkelo — APPROVED** [review]

File: contribute/deprecation-policy.md (hunk 8 above)
Comment (bergquist) [reply contribute/deprecation-policy.md]: Solved :)

**Review by timlevett — APPROVED** [review]
LGTM, thanks for putting this together.
